pkg/controller/cheinstallation/cheinstallation_controller.go

```diff
@@ -120,109 +119,97 @@ func (r *ReconcileCheInstallation) Reconcile(request reconcile.Request) (reconci
-		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, cheInstallation, r.setStatusCheSubscriptionFailed, err, "failed to create che subscription in namespace %s", cheInstallation.Spec.CheOperatorSpec.Namespace)
+		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, cheInstallation, r.setStatusCheInstallationFailed, err, "failed to create che subscription in namespace %s", cheInstallation.Spec.CheOperatorSpec.Namespace)
 	} else if created {
 		return reconcile.Result{}, nil
 	}
 
 	if requeue, err := r.ensureWatchCheCluster(); err != nil {
-		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, cheInstallation, r.setStatusCheSubscriptionFailed, err, "failed to add watch for CheCluster")
+		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, cheInstallation, r.setStatusCheInstallationFailed, err, "failed to add watch for CheCluster")
 	} else if requeue {
 		return reconcile.Result{Requeue: true, RequeueAfter: 3 * time.Second}, nil
 	}
 
 	if created, statusMsg, err := r.ensureCheCluster(reqLogger, cheInstallation); err != nil {
-		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, cheInstallation, r.setStatusCheSubscriptionFailed, err, "failed to create che cluster in namespace %s", cheInstallation.Spec.CheOperatorSpec.Namespace)
+		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, cheInstallation, r.setStatusCheInstallationFailed, err, "failed to create che cluster in namespace %s", cheInstallation.Spec.CheOperatorSpec.Namespace)
 	} else if created { // TODO VN: created can be removed
 		return reconcile.Result{}, nil
 	} else if statusMsg != "" {
-		return reconcile.Result{}, r.statusUpdate(reqLogger, cheInstallation, r.setStatusCheSubscriptionInstalling, statusMsg)
+		return reconcile.Result{}, r.statusUpdate(reqLogger, cheInstallation, r.setStatusCheInstallationInstalling, statusMsg)
 	}
 
-	return reconcile.Result{}, r.statusUpdate(reqLogger, cheInstallation, r.setStatusCheSubscriptionReady, "")
+	return reconcile.Result{}, r.statusUpdate(reqLogger, cheInstallation, r.setStatusCheInstallationSucceeded, "")
 }
 
 func (r *ReconcileCheInstallation) ensureCheNamespace(logger logr.Logger, cheInstallation *v1alpha1.CheInstallation) (bool, error) {
 	cheOpNamespace := cheInstallation.Spec.CheOperatorSpec.Namespace
-	ns := &v1.Namespace{}
-	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: cheOpNamespace}, ns); err != nil {
-		if errors.IsNotFound(err) {
-			logger.Info("Creating a namespace for che operator", "Namespace", cheOpNamespace)
-			namespace := NewNamespace(cheOpNamespace)
-			if err := controllerutil.SetControllerReference(cheInstallation, namespace, r.scheme); err != nil {
+	namespace := NewNamespace(cheOpNamespace)
+	if err := controllerutil.SetControllerReference(cheInstallation, namespace, r.scheme); err != nil {
+		return false, err
+	}
+	if err := r.client.Create(context.TODO(), namespace); err != nil {
```

Comment:
To me this seems to be an anti-pattern "always create" against current way "create if not exists".

Reply:
I understand your concern, but I assume the checking is done by the server, so there is no creation if a resource with the same name already exists.
The purpose of these functions are more straightforward to understand: they attempt to create the resource and treat the error accordingly. As a result, the complexity of the code is reduced. Also, I believe that trying to create immediately is more "atomic" in the sense that checking if the resource exists is done by the server.

But sure, it's a shift in term of approach...